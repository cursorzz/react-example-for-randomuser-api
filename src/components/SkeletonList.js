import React from 'react'
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Skeleton from '@material-ui/lab/Skeleton';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles({
  root: {
    display: "flex",
    alignItems: "center",
    '& div': {
      marginRight: 10

    }
  },
});

export function SkeletonList({ rows }) {
  const classes = useStyles();

  return (rows.map((row, index) => {
    return (
      <TableRow key={index}>
        <TableCell component="th" scope="row" className={classes.root}>
          <Skeleton variant="circle" width={40} height={40} />
          <Skeleton variant="text" width={"50%"} height={40} />
        </TableCell>
        <TableCell component="th" scope="row">
          <Skeleton variant="text" height={40} />
        </TableCell>
        <TableCell component="th" scope="row">
          <Skeleton variant="text" height={40} />
        </TableCell>
        <TableCell component="th" scope="row">
          <Skeleton variant="text" height={40} />
        </TableCell>
      </TableRow>)
  })
  )
}