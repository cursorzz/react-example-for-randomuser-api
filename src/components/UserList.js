import React from 'react'
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Avatar from '@material-ui/core/Avatar';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    display: "flex",
    alignItems: "center",
    '& div': {
      marginRight: 10

    }
  },
});

export function UserList({ rows }) {
  const classes = useStyles();

  const getName = (result) => {
    const { name } = result
    return `${name.title} ${name.first} ${name.last}`
  }

  return (rows.map(row => {
    return (<TableRow key={row.login.uuid}>
      <TableCell component="th" scope="row" className={classes.root}>
        <Avatar alt={getName(row)} src={row.picture.thumbnail} />
        {getName(row)}
      </TableCell>
      <TableCell align="left" component="th">{row.location.city}</TableCell>
      <TableCell align="left" component="th">{row.location.state}</TableCell>
      <TableCell align="left" component="th">{row.email}</TableCell>
    </TableRow>)
  }))
}