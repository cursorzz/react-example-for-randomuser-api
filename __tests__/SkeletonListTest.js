import React from "react";
import { SkeletonList } from "../src/components/SkeletonList";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(<SkeletonList rows={new Array(1).fill({})} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
