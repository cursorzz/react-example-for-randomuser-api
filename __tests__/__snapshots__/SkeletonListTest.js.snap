// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  className="MuiTableRow-root"
>
  <th
    aria-sort={null}
    className="MuiTableCell-root makeStyles-root-1"
    scope="row"
  >
    <div
      className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
    <div
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      style={
        Object {
          "height": 40,
          "width": "50%",
        }
      }
    />
  </th>
  <th
    aria-sort={null}
    className="MuiTableCell-root"
    scope="row"
  >
    <div
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      style={
        Object {
          "height": 40,
          "width": undefined,
        }
      }
    />
  </th>
  <th
    aria-sort={null}
    className="MuiTableCell-root"
    scope="row"
  >
    <div
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      style={
        Object {
          "height": 40,
          "width": undefined,
        }
      }
    />
  </th>
  <th
    aria-sort={null}
    className="MuiTableCell-root"
    scope="row"
  >
    <div
      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
      style={
        Object {
          "height": 40,
          "width": undefined,
        }
      }
    />
  </th>
</tr>
`;
